src/services/bundler/index.ts

```
@@ -356,6 +366,59 @@ export class BundlerService {
     return hash;
   }
 
+  async mintERC20Token(
+    signer: ethers.Signer,
+    tokenAddress: string,
+    from: string,
+    to: string,
+    amount: string,
+    description?: string
+  ): Promise<string> {
+    if (!isFunctionInABI("mint", tokenContractAbi)) {
+      throw new Error("Mint function not found in token contract ABI");
+    }
+
+    const formattedAmount = ethers.parseUnits(
+      amount,
+      this.config.token.decimals
+    );
+
+    const calldata = mintCallData(tokenAddress, to, formattedAmount);
+
+    const owner = await signer.getAddress();
+
+    let userop = await this.prepareUserOp(owner, from, calldata);
+
+    try {
+      // get the paymaster to sign the userop
+      userop = await this.paymasterSignUserOp(userop);
+
+      // sign the userop
+      const signature = await this.signUserOp(signer, userop);
+
+      userop.signature = signature;
+    } catch (e) {
+      throw new Error("Error preparing user op");
```

Potentially obfuscates the actual error in the library

---

Maybe rethrow the error so you are giving it some extra context?

```
} catch (e) {
      throw new Error(`Error preparing user op: ${e}`);
}
```